src/connectors/tidal.js

```diff
@@ -1,42 +1,31 @@
 'use strict';
 
-let useShortTrackNames = false;
+Connector.playerSelector = '#footerPlayer';
 
-readConnectorOptions();
+Connector.playButtonSelector = `${Connector.playerSelector} [data-type="button__pause"]`;
 
-Connector.playerSelector = '#nowPlaying';
-
-Connector.playButtonSelector = `${Connector.playerSelector} [class*="playbackToggle"]`;
+Connector.pauseButtonSelector = `${Connector.playerSelector} [data-type="button__play"]`;
 
 Connector.isScrobblingAllowed = () => !!$(Connector.playButtonSelector);
 
-Connector.isPlaying = () => $(Connector.playButtonSelector).attr('data-test') === 'pause';
-
-const fullTrackSelector = `${Connector.playerSelector} [class^="mediaInformation"] span:nth-child(1) a`;
-const shortTrackSelector = `${Connector.playerSelector} [class^="infoTableWrapper"] > table:nth-child(1) > tbody > tr:nth-child(1) > td:nth-child(2)`;
-
-Connector.getTrack = () => Util.getTextFromSelectors(useShortTrackNames ? shortTrackSelector : fullTrackSelector);
+Connector.trackSelector = `${Connector.playerSelector} [data-test="footer-track-title"] a[href^="/track/"]`;
 
 Connector.getUniqueID = () => {
-	const trackUrl = $(fullTrackSelector).attr('href');
+	const trackUrl = $(Connector.trackSelector).attr('href');
 	if (trackUrl) {
 		return trackUrl.split('/').pop();
 	}
 	return null;
 };
 
-Connector.artistSelector = `${Connector.playerSelector} [class^="mediaArtists"]`;
+Connector.artistSelector = `${Connector.playerSelector} .artist-link`;
 
-Connector.albumSelector = `${Connector.playerSelector} [class^="infoTable--"] a[href^="/album/"]`;
+Connector.albumSelector = ['#nowPlaying [class^="infoTable--"] a[href^="/album/"]', `${Connector.playerSelector} a[href^="/album"]`];
```

Comment:
The album information is no longer available in the info box. In fact, it's worse than it was when I first implemented the feature: The album name is only available if the expanded now playing box is open _and_ the credits tab is active. What I settled on is using the info table if it is available, but falling back to the "Playing From", though we ensure that the source is an album instead of a playlist. Let me know if the selector for that is overbroad.

Reply:
I tried narrowing this a bit. It seems like there's no other good way to do it.